day1/es3/Chapter.qml

```diff
@@ -0,0 +1,105 @@
+import QtQuick 2.0
+
+Column {
+    id: frame
+
+    property string title: "no tile"
+    property string content: "no content"
+    property bool isExpanded: false
+    property int animationTime: 2000
+    readonly property int contentMaxHeight: 200
+
+    signal clickChapter(var clickedChapter)
+
+    width: 160
+
+    Rectangle {
+        id: title
+
+        width: frame.width
+        height: 40
+        color: "gold"
+
+        Text {
+            anchors.centerIn: parent
+            text: frame.title
+            font.pointSize: 20
+        }
+
+        MouseArea {
+            anchors.fill: parent
+            onClicked: {
+                clickChapter(frame)
+            }
+        }
+    }
+
+    Rectangle {
+        id: content
+
+        width: frame.width
+        color: "deepskyblue"
+        clip:true
+
+        Text {
+            id: contentText
+
+            x: 0
+            text: frame.content
+            wrapMode: Text.WordWrap
+            width: content.width
+            height: contentHeight
+            horizontalAlignment: Text.AlignHCenter
+
+            Behavior on y {
```

Comment:
Per questo esiste un componente apposito che si chiama [YAnimator](https://doc.qt.io/qt-5/qml-qtquick-yanimator.html), è ottimizzato per questo tipo di effetto.

Questo in particolare non l'ho mai usato, ma penso sia adeguato.

Reply:
per la y ho usato una Transition, ho visto che utilizzando la transition, YAnimator dava problemi, quindi ho lasciato numberAnimation

Reply:
Può darsi che gli `Animator` diano dei problemi. Dalla [documentazione](https://doc.qt.io/qt-5/qml-qtquick-animator.html):
```
Animator types are a special type of animation which operate directly on Qt Quick's scene graph, rather than the QML objects and their properties like regular Animation types do. This has the benefit that Animator based animations can animate on the scene graph's rendering thread even when the UI thread is blocked.

The value of the QML property will be updated after the animation has finished. The property is not updated while the animation is running.
```

Quindi se hai qualche binding attivo sulla Y non ti si aggiorna fino alla fine dell'animazione.